spec/services/billing_entities/update_invoice_issuing_date_settings_service_spec.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe BillingEntities::UpdateInvoiceIssuingDateSettingsService do
+  include ActiveJob::TestHelper
+
+  subject(:update_service) { described_class.new(billing_entity:, params:) }
+
+  let(:billing_entity) { create(:billing_entity, invoice_grace_period: 9) }
+  let(:organization) { billing_entity.organization }
+  let(:customer) { create(:customer, organization:, net_payment_term: 5) }
+  let(:params) do
+    {
+      billing_configuration: {
+        subscription_invoice_issuing_date_anchor: "current_period_end",
+        subscription_invoice_issuing_date_adjustment: "align_with_finalization_date",
+        invoice_grace_period: 15
+      }
+    }
+  end
+
+  describe "#call" do
+    let!(:invoice_draft) do
+      create(
+        :invoice,
+        customer:,
+        billing_entity:,
+        status: :draft,
+        issuing_date: DateTime.parse("19 Jun 2022").to_date,
+        applied_grace_period: 9
+      )
+    end
+
+    context "with premium feature" do
+      around { |test| lago_premium!(&test) }
+
+      it "updates invoice issuing date settings on billing_entity" do
+        update_service.call
+
+        billing_entity.reload
+
+        expect(billing_entity.invoice_grace_period).to eq(15)
+        expect(billing_entity.subscription_invoice_issuing_date_anchor).to eq("current_period_end")
+        expect(billing_entity.subscription_invoice_issuing_date_adjustment).to eq("align_with_finalization_date")
+      end
+
+      it "updates issuing_date and payment_due_date on draft invoices" do
+        expect { perform_enqueued_jobs { update_service.call } }
+          .to change { invoice_draft.reload.issuing_date }.to(DateTime.parse("25 Jun 2022"))
+          .and change { invoice_draft.reload.payment_due_date }.to(DateTime.parse("30 Jun 2022"))
```

Comment:
We rarely perform jobs in the services as far as I know. It should be enough to ensure we schedule the job here.